src/commands/token/create-token.ts

```diff
@@ -26,54 +30,181 @@ export class CreateToken extends BaseCommand<typeof CreateToken> {
     'public-key-file': Flags.string({
       char: 'f',
       summary: 'Location of the public key PEM file',
+      required: false,
     }),
     algorithm: Flags.custom<SupportedAlgorithms>({
+      char: 'a',
       summary: 'The specific cryptographic algorithm used with the key',
       options: Object.values(SupportedAlgorithms),
       default: SupportedAlgorithms.RS256,
     })(),
+    quiet: Flags.boolean({
+      char: 'q',
+      default: false,
+      summary: 'Create ready-to-use PAT with auto-generated private public key pair and set its access policies',
+    }),
+    passphrase: Flags.string({
+      char: 'p',
+      summary: 'Passphrase for generation of private public key pair',
+      required: false,
+    }),
   }
 
   public async run(): Promise<TokenDto> {
     const { flags } = await this.parse(CreateToken)
-    const promptFlags = await promptRequiredParameters(['name', 'key-id', 'public-key-file'], flags)
+
+    await this.validateFlags(flags)
+
+    ux.action.start('Creating Personal Access Token')
+
+    const { token, publicKey, privateKey } = await this.createToken(flags)
+
+    await this.addPrincipal(token.id)
+    await this.updatePolicies(token.id)
```

Comment:
promise.all?

Reply:
Thanks. I've made a minor change, though both should not be in Promise.all since updatePolicies should be called after addPrincipal is completed.